virtool_cli/ref/repo.py

```diff
@@ -596,14 +533,114 @@ def _load_otu_events_from_cache_and_update(self, otu_id: uuid.UUID) -> list[int]
                 updated_events=otu_event_list,
             )
 
-            self._event_index_cache.cache_otu_events(
-                otu_id, otu_event_list, last_id=self.last_id
-            )
+            self._cache.cache_otu_events(otu_id, otu_event_list, last_id=self.last_id)
 
             return otu_event_list
 
         return []
 
+    def get_otu_metadata(self, event_ids: list[int]) -> dict | None:
+        """Retrieves OTU metadata from a list of event IDs"""
+        if not event_ids:
+            return None
+        event_ids.sort()
+        first_event_id = event_ids[0]
```

Comment:
Avoid mutating your function parameter.
```suggestion
        first_event_id = sorted(event_ids)[0]
```